### PR TITLE
Prac 06 code review request

### PR DESCRIPTION
**Pull request by Ronjn**
check for everything and anything you want @alexdamrow

### CODE REVIEW

File: prac_06/guitars.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    print("My guitars!")
+    name = input("Name: ")
+
+    while name != "":
+        year = int(input("Year: "))
+        cost = float(input("Cost: $"))
+
+        guitars.append(Guitar(name, year, cost))
+
+        print(f"{guitars[count]} added.\n")
```
Comment (alexdamrow) [inline prac_06/guitars.py:23]: Not sure you really need this print statement or the count variable as it not part of the question. But points to you for practicing using counting.

**Review by alexdamrow — COMMENTED** [review]
Nice work hard to find many errors with the code. Need to remember to include docstrings in your function to explain what they do, and also in  ___init___ to explain what the type the fields are and what they do.

File: prac_06/car.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""CP1404/CP5632 Practical - Car class example."""
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, fuel=0, name=""):
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
```
[the remark this answers is not in the document]
Comment (Ronjn) [reply prac_06/car.py:10]: Well spotted, fixed :).

File: prac_06/guitar.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    def __init__(self, name="", year=0, cost=0):
+        self.name = name
+        self.year = year
+        self.cost = cost
+
+    def __str__(self):
+        return f"{self.name}, ({self.year}) : ${self.cost}"
+
+    def get_age(self):
+        age = 2023 - self.year
```
[the remark this answers is not in the document]
Comment (Ronjn) [reply prac_06/guitar.py]: Good thinking, fixed :).

File: prac_06/guitar.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    def __init__(self, name="", year=0, cost=0):
+        self.name = name
+        self.year = year
+        self.cost = cost
+
+    def __str__(self):
+        return f"{self.name}, ({self.year}) : ${self.cost}"
+
+    def get_age(self):
```
[the remark this answers is not in the document]
Comment (Ronjn) [reply prac_06/guitar.py:26]: Thanks for pointing this out, fixed.

File: prac_06/guitar.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+"""
+Estimated time to complete  : 45  minutes
+Actual time to complete     : 60 minutes
+"""
+
+
+class Guitar:
+
+    def __init__(self, name="", year=0, cost=0):
+        self.name = name
```
[the remark this answers is not in the document]
Comment (Ronjn) [reply prac_06/guitar.py:19]: Fixed :)

File: prac_06/programming_language.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+class ProgrammingLanguage:
+
+    def __init__(self, name="", typing="", reflection="", year=0):
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def is_dynamic(self):
+        if self.typing == "Dynamic":
```
[the remark this answers is not in the document]
Comment (Ronjn) [reply prac_06/programming_language.py]: Thank you, I have learned something new :)

File: prac_06/programming_language.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+class ProgrammingLanguage:
+
+    def __init__(self, name="", typing="", reflection="", year=0):
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def is_dynamic(self):
```
[the remark this answers is not in the document]
Comment (Ronjn) [reply prac_06/programming_language.py:22]: Well spotted, thank you :)

File: prac_06/guitars.py (hunk 1 above)
Comment (Ronjn) [reply prac_06/guitars.py:23]: Yeah completely unnecessary but feels satisfying when testing the program to get this feedback haha.

**Pull request merged**